core/src/worker/workflow/history_update.rs

```diff
@@ -147,12 +152,13 @@ impl HistoryPaginator {
             npt,
             client,
         );
+        if empty_hist && wft.legacy_query.is_none() && wft.query_requests.is_empty() {
+            return Err(EMPTY_TASK_ERR.clone());
```

Comment:
Q: Does lang needs to expect and catch this error?

Reply:
Nope. You don't need to do anything special, it's just going to show up in evictions / logs